alacritty_terminal/src/term/search.rs

```diff
@@ -290,7 +285,11 @@ impl<T> Term<T> {
     ) {
         match direction {
             Direction::Right if cell.flags.contains(Flags::WIDE_CHAR) => {
-                iter.next();
+                // If advances `iter` at the end of column one character is skipped
+                // unintentionally.
```

Comment:
Having a comment on this makes sense, but the comment itself doesn't. This is not about the "end of column", but the end of the row (i.e. the last column in the row).

Something more like this seems more appropriate:

```
// In the alternate screen buffer there might not be a wide char spacer, so we only advance the iterator when the wide char is not in the last column.
```

A bit verbose maybe, but I think that is a better description of the issue.